builder.md

```diff
@@ -82,41 +87,68 @@ Each template repository MUST have a templates folder containing a config.yaml a
-- `base00-rgb-r` to `base0F-rgb-r` - red component as a value between `0` and `255`. e.g "124"
-- `base00-rgb-g` to `base0F-rgb-g` - green component as a value between `0` and `255`. e.g "175"
-- `base00-rgb-b` to `base0F-rgb-b` - blue component as a value between `0` and `255` e.g "194"
-- `base00-dec-r` to `base0F-dec-r` - red component as a value between `0` and `1.0`. e.g "0.4863"
-- `base00-dec-g` to `base0F-dec-g` - green component as a value between `0` and `1.0`. e.g "0.6863"
-- `base00-dec-b` to `base0F-dec-b` - blue component as a value between `0` and `1.0`. e.g "0.7608"
+- `scheme-name` - obtained from the `name` key of the scheme input
+- `scheme-author` - obtained from the `author` key of the scheme input
+- `scheme-description` - obtained from the `description` key of the scheme input
+- `scheme-slug` - obtained from the `slug` key of the scheme input (fallback value: a [slugified](#slugify) `scheme-name`)
+- `scheme-system` - obtained from the `system` key of the scheme input
+- `scheme-variant` - obtained from the `variant` key of the scheme input
+- `is_{{variant}}_variant` - dynamic value built from the `variant` key of the scheme file. e.g. `variant: "light"` provides `is_light_variant` with a value of `true`.
 
-## Considerations
+Additionally, a builder MUST provide the following template variables for each defined palette token:
+
+- `{{ token-name }}-hex` - 6-digit hex color value. This does not include a leading `#`. e.g "7cafc2".
+- `{{ token-name }}-hex-bgr` - built from a reversed version of all the hex values e.g "c2af7c"
+- `{{ token-name }}-hex-r` - red component of the hex color value. e.g "7c"
+- `{{ token-name }}-hex-g` - green component of the hex color value. e.g "af"
+- `{{ token-name }}-hex-b` - blue component of the hex color value. e.g "c2"
+- `{{ token-name }}-rgb-r` - red component as a value between `0` and `255`. e.g "124"
+- `{{ token-name }}-rgb-g` - green component as a value between `0` and `255`. e.g "175"
+- `{{ token-name }}-rgb-b` - blue component as a value between `0` and `255` e.g "194"
+- `{{ token-name }}-dec-r` - red component as a value between `0` and `1.0`. e.g "0.4863"
+- `{{ token-name }}-dec-g` - green component as a value between `0` and `1.0`. e.g "0.6863"
+- `{{ token-name }}-dec-b` - blue component as a value between `0` and `1.0`. e.g "0.7608"
+
+## Slugify
+
+Slugify is simplest to implement in a number of passes:
 
-Mustache was chosen as the templating language due to its simplicity and widespread adoption across languages. YAML was chosen to describe scheme and configuration files for the same reasons.
+* Start with your input value, replacing any Unicode characters with their ASCII aproximations (as an example é would become e). On a technical level, this can be done by normalizing the string to the Unicode NFD form (which is the decomposed version), and then dropping any combining characters.
```

Comment:
```suggestion
* Start with your input value, replacing any Unicode characters with their ASCII aproximations (as an example `é` would become `e`). On a technical level, this can be done by normalizing the string to the Unicode NFD form (which is the decomposed version), and then dropping any combining characters.
```